0014-fungible-token-standard.md

```diff
@@ -0,0 +1,98 @@
+# Fungible Token Standard on Mina
+
+## Summary
+
+Fungible tokens are a pervasive feature that are used for many applications, such as bridges, decentralised exchanges, layer 2 solutions, and others. Establishing a standard for creating fungible tokens on Mina will reduce the efforts for builders to create their own tokens, and allow exchanges, explorers, and other third party applications to integrate fungible tokens in a uniform way.
+
+## Motivation
+
+Fungible tokens are a common feature on programmable blockchain systems. Establishing a standard will help developers to create their own fungible tokens, and to interface their application with tokens that others have created.
+
+We propose a modular standard, consisting of a number of `TypeScript` interfaces. These define an API which third parties can target for integration.  We also provide a reference implementation, based on the [custom token feature](https://github.com/MinaProtocol/MIPs/blob/main/MIPS/mip-zkapps.md#custom-tokens) of Mina and the [`TokenContract` class](https://github.com/o1-labs/o1js/pull/1384) of `o1js.`
+
+The standard supports the common features associated with fungible tokens: minting and burning, transferring, and viewing account balances. Transfers can be initiated by either calling a `@method` of the contract defining the token (the _token owner contract_), or by having `AccountUpdate`s which contain balance changes _approved_ by the token owner contract. In particular, this design allows interoperability in the sense that arbitrary third party `SmartContract`s can use fungible tokens.
+
+## Detailed design
+
+Fungible tokens on Mina will use the custom token feature of Mina defined in [MIP4](https://github.com/MinaProtocol/MIPs/blob/main/MIPS/mip-zkapps.md#custom-tokens). In Mina, a new class of custom tokens can be introduced by writing and deploying a `SmartContract` that defines the token. This is called the _token owner contract_. When the token owner contract is deployed, a new token id is created, and accounts with that token id will hold the custom token instead of Mina.
+
+The token owner contract can change balances of accounts with the custom token, so it can mint, burn, and move tokens. When an `AccountUpdate` that has not been created in a method of the owner contract tries to modify the balance of an account with the custom token, it needs to be approved by the owner contract. That way, the owner contract can enforce rules that all transactions with the token should satisfy (conservation of tokens being a very common example). The [`TokenContract` class](https://github.com/o1-labs/o1js/pull/1384) defines methods for token transfer, as well as for approving a whole forest of account updates. The API that we define mirrors those methods, and the reference implementation is built using the `TokenContract`.
+
+The API consists of the following `TypeScript` `interface`s:
```

Comment:
I'll let others closer to the code comment also, but these interfaces look pretty clean and understandable to met @kantp 👍